Sources/AppStoreConnectCLI/Commands/TestFlight/BetaTesters/ListBetaTestersCommand.swift

```diff
@@ -0,0 +1,106 @@
+            help: "Number of included related resources to return.")
+    var relatedResourcesLimit: Int
+
+    private enum ListBetaTesterError: LocalizedError {
+        case multipleFilters
+
+        var failureReason: String? {
+            switch self {
+                case .multipleFilters:
+                    return "Only one relationship filter can be applied"
+            }
+        }
+    }
+
+    private enum ListStrategy {
+        case all
+        case listByApp(bundleId: String)
+        case listByGroup(betaGroupName: String)
+        case listByAppAndGroup
+
+        typealias ListOptions = (bundleId: String?, betaGroupName: String?)
+
+        init(options: ListOptions) {
+            switch (options.bundleId, options.betaGroupName) {
+                case let(.some(bundleId), .some(betaGroup)) where !bundleId.isEmpty && !betaGroup.isEmpty:
+                    self = .listByAppAndGroup
+                case let(.some(bundleId), _) where !bundleId.isEmpty:
+                    self = .listByApp(bundleId: bundleId)
+                case let(_, .some(betaGroupName)) where !betaGroupName.isEmpty:
+                    self = .listByGroup(betaGroupName: betaGroupName)
+                case (_, _):
+                    self = .all
+            }
+        }
+    }
+
+    func run() throws {
+        let api = try makeClient()
+
+        let request: AnyPublisher<BetaTestersResponse, Error>
+
+        let includes = [ListBetaTesters.Include.apps, ListBetaTesters.Include.betaGroups]
+        let limits = [ListBetaTesters.Limit.apps(relatedResourcesLimit),
+                      ListBetaTesters.Limit.betaGroups(relatedResourcesLimit)]
+
+        switch ListStrategy(options: (filterBundleId, filterGroupName)) {
+            case .all:
+                request = api
+                    .request(APIEndpoint.betaTesters(include: includes, limit: limits))
+                    .eraseToAnyPublisher()
+
+            case .listByApp(let bundleId):
+                request = api
+                    .getAppResourceIdsFrom(bundleIds: [bundleId])
+                    .flatMap {
+                        api.request(APIEndpoint.betaTesters(
+                            filter: [.apps($0)],
+                            include: includes,
+                            limit: limits
+                        ))
+                    }
+                    .eraseToAnyPublisher()
+
+            case .listByGroup(let betaGroupName):
+                request = try api.betaGroupIdentifier(matching: betaGroupName)
+                    .flatMap {
+                        api.request(APIEndpoint.betaTesters(
+                            filter: [.betaGroups([$0])],
+                            include: includes,
+                            limit: limits
+                        ))
+                    }
+                    .eraseToAnyPublisher()
+            case .listByAppAndGroup:
+                throw ListBetaTesterError.multipleFilters
```

Comment:
Let's make every all paths go through the completion handler
Instead of throwing from here we can set the request to be a `Fail` publisher rather than exiting early